### PR TITLE
[feat] implement create application constraints

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Extracted away logic to continue an incomplete application
- Added logic to check constraints when creating a new application

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Create conditions for the following and verify that you cannot create a new application:
- Have an application that was rejected
- Have an application that is pending/incomplete
- Have an application that was accepted but submitted within 6 months
- Have an invalid ending reason for past correspondences

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Implement preemptive check + disable create new app button
- Implement toast notification for unexpected errors

CC: @InayaY3

### CODE REVIEW

File: actions/applications/continueApplication.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return {
+      data: null,
+      error: `Couldn't find application with id ${appId}`,
+    };
+  }
+
+  return {
+    data: appData,
+    error: null,
+  };
```
Comment (InayaY3) [inline actions/applications/continueApplication.ts]: will this be passing input into `getResumeStageAndQuestion`? meaning that we should be calling it in `MainDashboard/ApplicationCardButton.tsx`? given that you should be able to route to an incomplete application by clicking on it (and not the new application button anymore)

File: actions/applications/createApplication.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import { getSupabaseServerClient } from '@/lib/supabase';
+import { AdopterApplication } from '@/types/schema';
+import Logger from '../logging';
+
+/**
+ * Checks the requesting user's ability to create
+ * a new application. Returns true if the user is
+ * able to create, false otherwise.
+ */
+export const checkCreationConstraints = async (user: User) => {
```
Comment (InayaY3) [inline actions/applications/createApplication.ts]: i also don't see any checks for the "2 or more active adoptees already" case. can you add that, and check both the adopter applications table (which you're already accessing for the 6 month check) and the profile table for `num_past_active` (which you'll also be checking for the invalid ending reason)

i know the `num_past_active` col is not the best implementation given that we could try checking against monday, but let's have this as a placeholder for now. being able to successfully parse monday for past adoptees seems unlikely (the npo seems to kind of agree). have some other ideas on how to tackle it but again, just for now let's add this check.
